### PR TITLE
@types/nested-error-stacks Added typings for nested-error-stacks

### PR DESCRIPTION
**Pull request by woutervh-**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by woutervh-** [comment]
Fixing some issue with accessing node types.

**Comment by woutervh-** [comment]
Fixed remaining issues.

File: types/nested-error-stacks/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+// Type definitions for nested-error-stacks 2.1
+// Project: https://github.com/mdlavin/nested-error-stacks
+// Definitions by: Wouter van Heeswijk <https://github.com/woutervh->
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export default class NestedError extends Error {
```
Comment (armanio123) [inline types/nested-error-stacks/index.d.ts]: It seems like the library not using `export default`, so this .d.ts file should not either.

Instead, you should follow the appropriate template from [the TypeScript handbook](http://www.typescriptlang.org/docs/handbook/declaration-files/templates.html).

File: types/nested-error-stacks/nested-error-stacks-tests.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import NestedErrorStacks from 'nested-error-stacks';
```
Comment (armanio123) [inline types/nested-error-stacks/nested-error-stacks-tests.ts]: You should use `import name = require('package');` instead.

If you still want to use the syntax `import name from 'package';` you need to use `esModuleInterop` on the compiler settings from your consuming code.

**Review by armanio123 — CHANGES_REQUESTED** [review]

**Comment by woutervh-** [comment]
Thanks @armanio123. I have changed the export/import according to your suggestion.

**Review by armanio123 — APPROVED** [review]

**Pull request merged**